src/components/ColorList/ColorList.tsx

```diff
@@ -0,0 +1,33 @@
+import { IColor } from "../../interfaces";
+import './ColorList.scss';
+
+type ColorListProps = {
+  colors: IColor[],
+  maxItemsVisible?: number,
+  onChangeColor?: (colorItem: IColor) => void
+}
+
+const ColorList: React.FC<ColorListProps> = ({ colors, maxItemsVisible, onChangeColor }) => {
+  const isMaxItem = colors.length > Number(maxItemsVisible);
+
+  if (!colors.length) return null;
```

Comment:
if you swap these lines(11, 13), then you can avoid unnecessary checking when the color array is empty. Btw its not critical performance issure 💥💥💥